web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -0,0 +1,189 @@
+        webView?.goBack()
+    }
+
+    AndroidView(
+        factory = { context ->
+            WebView(context).apply {
+                onCreated(this)
+
+                webViewClient = object : WebViewClient() {
+                    override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+                        super.onPageStarted(view, url, favicon)
+                        state.isLoading = true
+                    }
+
+                    override fun onPageFinished(view: WebView?, url: String?) {
+                        super.onPageFinished(view, url)
+                        state.isLoading = false
+                        canGoBack = view?.canGoBack() ?: false
+                    }
+
+                    override fun doUpdateVisitedHistory(
+                        view: WebView?,
+                        url: String?,
+                        isReload: Boolean
+                    ) {
+                        super.doUpdateVisitedHistory(view, url, isReload)
+                        // WebView will often update the current url itself.
+                        // This happens in situations like redirects and navigating through
+                        // history. We capture this change and update our state holder url.
+                        // On older APIs (28 and lower), this method is called when loading
+                        // html data. We don't want to update the state in this case as that will
+                        // overwrite the html being loaded.
+                        if (url != null &&
+                            !url.startsWith("data:text/html") &&
+                            state.content.getCurrentUrl() != url
+                        ) {
+                            state.content = WebContent.Url(url)
+                        }
+                    }
+
+                    override fun onReceivedError(
+                        view: WebView?,
+                        request: WebResourceRequest?,
+                        error: WebResourceError?
+                    ) {
+                        super.onReceivedError(view, request, error)
+                        onError(request, error)
+                    }
+
+                    override fun shouldOverrideUrlLoading(
+                        view: WebView?,
+                        request: WebResourceRequest?
+                    ): Boolean {
+                        // Override all url loads to make the single source of truth
+                        // of the URL the state holder Url
+                        request?.let {
+                            val content = WebContent.Url(it.url.toString())
+                            state.content = content
+                        }
+                        return true
+                    }
+                }
+            }.also { webView = it }
+        },
+        modifier = modifier
+    ) { view ->
+        when (val l = state.content) {
```

Comment:
Nit: what's `val l` in this context? Can we be a bit more explicit with the val name?